src/taipy/core/_manager/_manager.py

```diff
@@ -44,7 +48,12 @@ def _delete_many(cls, ids: Iterable):
         cls._repository._delete_many(ids)
         if hasattr(cls, "_EVENT_ENTITY_TYPE"):
             for entity_id in ids:
-                _publish_event(cls._EVENT_ENTITY_TYPE, entity_id, EventOperation.DELETION, None)  # type: ignore
+                _publish_event(
+                    cls._EVENT_ENTITY_TYPE,  # type: ignore
+                    EventOperation.DELETION,
+                    entity_id=entity_id,
+                    delete_all=True,
```

Comment:
*Note to reviewers*: I think we should gradually replace these `_publish_events` my a more readable and (more important) more centralized `Notifier.publish(make_event(entity))`. 
what do you think?

Reply:
I don't have strong opinion on it but I don't believe it is more readable. It is not less readable either. Just a bit longer to me.

The method `_publish_event` makes the event and publishes it in a single call, while your proposal is to split it into 2 calls.

We could make the `_publish_event` a single dispatch method directly to apply your changes (passing the entity directly to add more fields and metadata) and keep a the two steps in a single call.

It's just a minor suggestion, but not at all a deal breaker. It's up to you to keep it as it is or not.

Reply:
The problem I point at is not the fact that there is two calls instead of one (we can make it a single call if you want to).
The problem is that the caller of `_publish_event` must currently implement all the logic: how to get the ID, how to get the metadata/extra information for the event etc.
As a consequence, this logic (which is very entity related) is scattered all over the code base. Just search for all the calls for `_publish_event` and you get an idea of what needs to change if you want to add just one more field!
My proposal is to make it more centralized and generic.
My two cents.